UPGRADE.md

```diff
@@ -63,6 +63,45 @@ import { jsonServerRestClient } from 'admin-on-rest';
 import jsonServerRestClient from 'ra-data-json-server';
 ```
 
+## Introduced `i18nProvider` for loading of locale messages
+
+Previously you would supply `react-admin` with a `messages` prop object to allow alternative locales, this should be replaced with an `i18nProvider`. The new `i18nProvider` allows you to load the messages asynchronously. @See [`i18nProvider`](./Translation.md#i18nProvider) 
+
+```jsx
+// before
+import { enMessages } from 'admin-on-rest';
+const messages = { 'en': enMessages };
```

Comment:
the two examples are not the same, since you're passing one language here and two in the second snippet. An upgrade guide should show how to achieve *the same thing* on the two versions.